### DISCUSSION TITLE
Vite SSR V1 - Handling FrontMatter exports with the new V1 design?

### DISCUSSION THREAD

**Discussion by graphite-admin** [vikejs/vike#869] [Discussion] [answered]
I migrated to V1 based on the advice given in [this issue](https://github.com/brillout/vite-plugin-ssr/issues/864). V1 migration solved the problem described in the issue, but the project I am working on uses MDX files. These mdx files export some front matter. So now I'm getting these errors:

```
[vite-plugin-ssr@0.4.123][Warning] /pages/+Page.mdx should only have a default export: remove `export { frontmatter }`
```

What's the best way to approach solving this? Ideally there should be some way to access the front matter defined on a mdx file.

**Comment by brillout** [comment]
That’a a good point. I’ll implement a solution later today.

**Reply by brillout** [reply, marked as answer]
In the meantime you can use this workaround:

```js
// /pages/some-page/+config.js

import Page, { frontmatter} from './Page.mdx'

export default {
  Page,
  frontmatter
}
```

Create a new config `frontmatter` using [`meta`](https://vite-plugin-ssr.com/meta). Let me know if you have questions about `meta` (the docs for `meta` are work-in-progress and are going to be improved in the coming days).

The DX of the workaround isn't ideal, so I'll still implement a solution. Let me know if the workaround works so that I know whether I've more time to think of a solution.

**Reply by graphite-admin** [reply]
Thanks for coming back to this so quickly. That work around works fine. Take your time with the fix/solution. 

If it helps, I'd be willing to help contribute once you determine the best way this should work

**Reply by brillout** [reply]
> Take your time with the fix/solution.

Thanks and I'm actually taking some time for it as I'm currently refactoring internals.

> If it helps, I'd be willing to help contribute once you determine the best way this should work

I think, for this one, it's best I fix it. That said, contribution much welcome on [contribution-welcome tickets](https://github.com/brillout/vite-plugin-ssr/labels/contribution-welcome%20%3Atwo_hearts%3A), or PM me on Discord if you'd rather take a stab at other things (potentially on other projects).

**Reply by brillout** [reply]
@graphite-admin Can you try with `0.4.124`? It should work in principe, but I didn't test it (I may have missed some small implementation details).
